Core/include/Acts/EventData/TrackStateProxy.hpp

```diff
@@ -615,145 +597,101 @@ class TrackStateProxy {
-              subspace.begin());
-    return subspace;
-  }
-
-  void setBoundSubspaceIndices(BoundSubspaceIndices boundSubspace)
+  /// Set the projector subspace indices
+  /// @param boundSubspace The projector subspace indices to set
+  void setProjectorSubspaceIndices(BoundSubspaceIndices boundSubspace)
     requires(!ReadOnly)
   {
     assert(has<hashString("projector")>());
     component<SerializedSubspaceIndices, hashString("projector")>() =
         serializeSubspaceIndices(boundSubspace);
   }
 
+  /// Set the projector subspace indices
+  /// @param subspace The projector subspace indices to set
   template <std::size_t measdim>
-  void setSubspaceIndices(SubspaceIndices<measdim> subspace)
+  void setProjectorSubspaceIndices(SubspaceIndices<measdim> subspace)
     requires(!ReadOnly && measdim <= eBoundSize)
   {
     assert(has<hashString("projector")>());
     BoundSubspaceIndices boundSubspace{};
     std::copy(subspace.begin(), subspace.end(), boundSubspace.begin());
-    setBoundSubspaceIndices(boundSubspace);
+    setProjectorSubspaceIndices(boundSubspace);
   }
 
+  /// Set the projector subspace indices
+  /// @param subspaceIndices The projector subspace indices to set
   template <std::size_t measdim, typename index_t>
-  void setSubspaceIndices(std::array<index_t, measdim> subspaceIndices)
+  void setProjectorSubspaceIndices(std::span<index_t, measdim> subspaceIndices)
     requires(!ReadOnly && measdim <= eBoundSize)
   {
     assert(has<hashString("projector")>());
     BoundSubspaceIndices boundSubspace{};
     std::transform(subspaceIndices.begin(), subspaceIndices.end(),
                    boundSubspace.begin(),
                    [](index_t i) { return static_cast<std::uint8_t>(i); });
-    setBoundSubspaceIndices(boundSubspace);
+    setProjectorSubspaceIndices(boundSubspace);
+  }
+
+  /// Set the projector subspace indices
+  /// @param subspaceIndices The projector subspace indices to set
+  template <typename index_t>
+  void setProjectorSubspaceIndices(std::span<index_t> subspaceIndices) {
+    assert(has<hashString("projector")>());
+    assert(subspaceIndices.size() <= eBoundSize);
+    BoundSubspaceIndices boundSubspace{};
+    std::transform(subspaceIndices.begin(), subspaceIndices.end(),
+                   boundSubspace.begin(),
+                   [](index_t i) { return static_cast<std::uint8_t>(i); });
+    setProjectorSubspaceIndices(boundSubspace);
+  }
+
+  /// Set the projector subspace indices
+  /// @param subspaceIndices The projector subspace indices to set
+  template <std::size_t measdim, typename index_t>
+  void setProjectorSubspaceIndices(std::array<index_t, measdim> subspaceIndices)
+    requires(!ReadOnly && measdim <= eBoundSize)
+  {
+    setProjectorSubspaceIndices(std::span(subspaceIndices));
+  }
+
+  /// Returns whether a projector is set
+  /// @return Whether it is set
+  bool hasProjector() const { return has<hashString("projector")>(); }
```

Comment:
Just to confirm, this is unchanged right?

Reply:
Yes this is still the same